### ISSUE TITLE
error upgrading 17.0.2 to 18.0.4: "Repair error: Repair step 'OCA\Talk\Migration\FixNamespaceInDatabaseTables' is unknown"

### ISSUE THREAD

**Issue by yzal67** [nextcloud/spreed#3654]
## Steps to reproduce
1. Just run /var/scripts/update.sh and got it

## Expected behaviour
Expected successful upgrade
## Actual behaviour
Upgrade failed

## Server configuration

**Nextcloud server version:** (see your admin page) 17.0.2

**Server OS** (Ubuntu server is default) Ubuntu 18.04.4 LTS

**How did you install the VM?** (Scripted install from master **OR** Released version) scripted install

## Network
**Do you use DHCP?** yes

**Is port 80 and/or 443 open?** yes

#### Log file (/var/log/nextcloud/nextcloud.log)
```
Insert your nextcloud.log file here
```

#### Installation log (command output)
(begin)
Repair error: Repair step 'OCA\Talk\Migration\FixNamespaceInDatabaseTables' is unknown
PHP Fatal error:  Cannot declare class OCA\Talk\Migration\Version8000Date20200331144101, because the name is already in use in /var/www/nextcloud/apps/spreed/lib/Migration/Version8000Date20200331144101.php on line 54
Sorry but something went wrong. Please report this issue to https://github.com/nextcloud/vm/issues and include the output of the error message. Thank you!
sudo -u www-data php /var/www/nextcloud/occ upgrade failed
Posting notification to users that are admins, this might take a while...
Posting 'Sorry but something went wrong. Please report this issue to https://github.com/nextcloud/vm/issues and include the output of the error message. Thank you!' to: nxadmin
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade

                                                                  
  There are no commands defined in the "notification" namespace.  
                                                                  
(end)

**Comment by nickvergessen** [comment]
Just uninstall and install the Talk app again.
This is sadly not easily fixable but the workaround is quite easy.
From Nextcloud 19 onwards it's fixed as the server doesn't load Talk files directly anymore before the app is loaded.

**Comment by kapitan-iglu** [comment]
Today I run into same issue. After upgrading NextCloud from 17.0.10 to 18.0.12 I'm unable to install Talk app anymore. 
`occ upgrade` ended with:
```
$ sudo -u www-data php7.3 occ upgrade
...
Checking for update of app spreed in appstore
Update app spreed from appstore
Repair error: Repair step 'OCA\Talk\Migration\FixNamespaceInDatabaseTables' is unknown
PHP Fatal error:  Cannot declare class OCA\Talk\Migration\Version8000Date20200331144101, because the name is already in use in /var/www/owncloud/apps/spreed/lib/Migration/Version8000Date20200331144101.php on line 54
```
So I removed `spreed` folder from `owncloud/apps/` and rerun upgrade process.
Afterwards I tried to install Talk from web admin interface and also from console several times. Still same error:
```
PHP Fatal error:  Cannot declare class OCA\Talk\Migration\Version8000Date20200331144101, because the name is already in use in /var/www/owncloud/apps/spreed/lib/Migration/Version8000Date20200331144101.php on line 54
```
After running `sudo -u www-data php7.3 occ app:install spreed` I found In NextCloud logs three errors:

```
[no app in context] Error: OCP\AppFramework\QueryException: Could not resolve OCA\Talk\AppInfo\Application! Class OCA\Talk\AppInfo\Application does not exist at <<closure>>

 0. /var/www/owncloud/lib/private/AppFramework/Utility/SimpleContainer.php line 115
    OC\AppFramework\Utility\SimpleContainer->resolve("OCA\\Talk\\AppInfo\\Application")
 1. /var/www/owncloud/lib/private/ServerContainer.php line 150
    OC\AppFramework\Utility\SimpleContainer->query("OCA\\Talk\\AppInfo\\Application", true)
 2. /var/www/owncloud/apps/spreed/appinfo/app.php line 22
    OC\ServerContainer->query("OCA\\Talk\\AppInfo\\Application")
 3. /var/www/owncloud/lib/private/legacy/app.php line 266
    require_once("/var/www/ownclo ... p")
 4. /var/www/owncloud/lib/private/legacy/app.php line 155
    OC_App::requireAppFile("spreed")
 5. /var/www/owncloud/lib/private/legacy/app.php line 966
    OC_App::loadApp("spreed")
 6. /var/www/owncloud/lib/private/Installer.php line 145
    OC_App::executeRepairSteps("spreed", {step: "OCA\\Tal ... "})
 7. /var/www/owncloud/core/Command/App/Install.php line 66
    OC\Installer->installApp("spreed")
 8. /var/www/owncloud/3rdparty/symfony/console/Command/Command.php line 255
    OC\Core\Command\App\Install->execute(Symfony\Componen ... {}, Symfony\Componen ... {})
 9. /var/www/owncloud/3rdparty/symfony/console/Application.php line 915
    Symfony\Component\Console\Command\Command->run(Symfony\Componen ... {}, Symfony\Componen ... {})
10. /var/www/owncloud/3rdparty/symfony/console/Application.php line 272
    Symfony\Component\Console\Application->doRunCommand(OC\Core\Command\App\Install {}, Symfony\Componen ... {}, Symfony\Componen ... {})
11. /var/www/owncloud/3rdparty/symfony/console/Application.php line 148
    Symfony\Component\Console\Application->doRun(Symfony\Componen ... {}, Symfony\Componen ... {})
12. /var/www/owncloud/lib/private/Console/Application.php line 214
    Symfony\Component\Console\Application->run(Symfony\Componen ... {}, Symfony\Componen ... {})
13. /var/www/owncloud/console.php line 99
    OC\Console\Application->run()
14. /var/www/owncloud/occ line 11
    require_once("/var/www/owncloud/console.php")

at 2021-01-09T13:28:29+00:00
```

```
[no app in context] Error: Exception: Repair step 'OCA\Talk\Migration\FixNamespaceInDatabaseTables' is unknown at <<closure>>

0. /var/www/owncloud/lib/private/legacy/app.php line 974
   OC\Repair->addStep("OCA\\Talk\\Migr ... s")
1. /var/www/owncloud/lib/private/Installer.php line 145
   OC_App::executeRepairSteps("spreed", {step: "OCA\\Tal ... "})
2. /var/www/owncloud/core/Command/App/Install.php line 66
   OC\Installer->installApp("spreed")
3. /var/www/owncloud/3rdparty/symfony/console/Command/Command.php line 255
   OC\Core\Command\App\Install->execute(Symfony\Componen ... {}, Symfony\Componen ... {})
4. /var/www/owncloud/3rdparty/symfony/console/Application.php line 915
   Symfony\Component\Console\Command\Command->run(Symfony\Componen ... {}, Symfony\Componen ... {})
5. /var/www/owncloud/3rdparty/symfony/console/Application.php line 272
   Symfony\Component\Console\Application->doRunCommand(OC\Core\Command\App\Install {}, Symfony\Componen ... {}, Symfony\Componen ... {})
6. /var/www/owncloud/3rdparty/symfony/console/Application.php line 148
   Symfony\Component\Console\Application->doRun(Symfony\Componen ... {}, Symfony\Componen ... {})
7. /var/www/owncloud/lib/private/Console/Application.php line 214
   Symfony\Component\Console\Application->run(Symfony\Componen ... {}, Symfony\Componen ... {})
8. /var/www/owncloud/console.php line 99
   OC\Console\Application->run()
9. /var/www/owncloud/occ line 11
   require_once("/var/www/owncloud/console.php")

at 2021-01-09T13:28:29+00:00
```
```
[PHP] Error: Cannot declare class OCA\Talk\Migration\Version8000Date20200331144101, because the name is already in use at /var/www/owncloud/apps/spreed/lib/Migration/Version8000Date20200331144101.php#54

at 2021-01-09T13:28:31+00:00
```

Solution to uninstall and install the Talk app again (found here and also in #2806) is not working for me. **Why?**

**Comment by nickvergessen** [comment]
> So I removed spreed folder from owncloud/apps/ and rerun upgrade process.
Afterwards I tried to install Talk from web admin interface and also from console several times. Still same error:

The problem is the upgrade will break the state again. Either disable the app before the upgrade, or afterwards manually remove it again and install it all again manually.

**Comment by kapitan-iglu** [comment]
> Either disable the app before the upgrade, or afterwards manually remove it again and install it all again manually.

0) Initial state: `spreed` folder exists inside apps/, webadmin offers me "Remove" or "Enable" (because of last installation fails).
1) Delete folder `spreed` from filesystem.
2) Reload webadmin and "Remove" application Talk.
3) Rerun upgrade:
  ```
$ sudo -u www-data php7.3 occ upgrade
Nextcloud is already latest version
```
4) Installation of Talk app from webadmin fails with same two errors.
5) Delete folder `spreed` from filesystem again.
6) Install app using occ:
  ```
$ sudo -u www-data php7.3 occ app:install spreed
spreed installed
spreed enabled
```
7) Talk is now available.

It's interesting that installing from webadmin fails but from console succeeds. Nevermind, thanks!

> So I removed spreed folder from owncloud/apps/ and rerun upgrade process.

My mistake was that I did not run `occ app:disable spreed` after I removed it from filesystem just before I run `occ upgrade` second time?
